src/htsp_server.c

```diff
@@ -999,6 +999,15 @@ htsp_build_dvrentry(htsp_connection_t *htsp, dvr_entry_t *de, const char *method
     htsmsg_add_u32(out, "priority",    u32);
     htsmsg_add_u32(out, "contentType", de->de_content_type);
 
+    //To not risk breaking older clients, only
+    //provide the 'age rating' via HTSP if the requested
+    //API version if greater than 35.
+    if (htsp->htsp_version > 35)
```

Comment:
Not sure this is the correct way to implement versioning. I think htsp version needs to be increased, so that a client after connecting to tvh knows the htsp version of the server is high enough so the client can expect and use the new property. 

Iirc, this is how we did former times.

Reply:
> Not sure this is the correct way to implement versioning. I think htsp version needs to be increased, so that a client after connecting to tvh knows the htsp version of the server is high enough so the client can expect and use the new property.
> 
> Iirc, this is how we did former times.

Thanks.  I will look into this.

Reply:
@ksooo - I looked into this:

When initiating contact via HTSP, the client sends a ‘hello’ message which includes the maximum HTSP version that it can understand.

```
{'clientname': 'My client name',
 'htspversion': 33,
 'seq': 1,
 'method': 'hello'}
```

When TVH receives the ‘hello’ message, it replies with an acknowledgement message that contains some details about the system including, but not limited to, the maximum HTSP version that it can accept.

```
[{'htspversion': 36,
'servername': 'Tvheadend',
'serverversion': '0.0.0-unknown',
'challenge': b'<SNIP>,
'language': 'eng',
'servercapability': ['caclient',
                     'tvadapters',
                     'satip_client',
                     'satip_server',
                     'trace',
                     'libav'],
'api_version': 19,
'seq': 1}]
```

The server then takes the lowest version number of either the client-requested version or the maximum supported version and sets the response protocol version to that value.  As far as I can tell, the server never actually confirms back to the client the active protocol version it is using.

https://github.com/tvheadend/tvheadend/blob/62adbebfd062d7b97829268274aad92df2033784/src/htsp_server.c#L1387-L1388

There are numerous areas within the module that test the active protocol version when formulating a response.

```
if (htsp->htsp_version < nn)
  set value A;
else
  set value B;
```

I can see that in Sep-2021, the version was incremented from 35 to 36 when the RDS field was added.  However, as far as I can tell, this version number is never tested for.  I suspect that regardless of the version that the client requests (Kodi pvr.hts is currently asking for v35), the ‘rds_uecp’ field will always be present.

I will soon be submitting a PR for my new 'rating label' feature.  I will increment the version number for that, but also include this field under that version.